Components/Sensors/MQ135.cs

```diff
@@ -0,0 +1,113 @@
+        public float GetReadingUncalibrated() => CalculatePPM(this.Input, this.RS, this.RL, this.SupplyVoltage, this.ResistanceCal, false, 0, 0);
+
+        /// <summary> Applies some basic calibration to the sensor to compensate for air temperature. </summary>
+        /// <remarks> Defaults to 20 C. </remarks>
+        /// <param name="Temperature"> The current air temperature in degrees C, between -10 and 45. </param>
+        [Obsolete("This does not yet apply any calibration.")]
+        public void CalibrateTemperature(float Temperature)
+        {
+            if (Temperature < -10 || Temperature > 45) { throw new Exception("MQ135 only works in -10 to 45 C temperatures."); }
+            this.Temperature = Temperature;
+        }
+
+        /// <summary> Applies some basic calibration to the sensor to compensate for environmental humidity. </summary>
+        /// <remarks> Defaults to 65% RH. </remarks>
+        /// <param name="Humidity"> % Relative Humidity value, between 0.0 and 0.95. </param>
+        [Obsolete("This does not yet apply any calibration.")]
+        public void CalibrateHumidity(float Humidity)
+        {
+            if (Humidity < 0 || Humidity > 0.95) { throw new Exception("MQ135 only works in 0 to 95% RH environment."); }
+            this.Humidity = Humidity;
+        }
+
+        /// <summary> Applies calibration for base sensor resistance. This is probably constant for each individual sensor. </summary>
+        /// <remarks> You'll need to put the sensor into a 100 ppm CO2 environment at 20C and 65% RH to get this value. 41763 is used as default. </remarks>
+        /// <param name="Resistance"> The measured resistance of the sensor, in the specific calibration conditions as listed above. Should be near 42K. </param>
+        public void CalibrateResistance(int Resistance) { this.ResistanceCal = Resistance; }
+
+        public static float CalculatePPM(IAnalogueIn Input, double RS, double RL, double SupplyVoltage, int ResistanceCal, bool UseCal, float Humidity, float Temperature)
+        {
+            double SensorResistance = ((RL * SupplyVoltage) / Input.GetInput()) - (RS + RL);
+            double ResistanceRatio = SensorResistance / ResistanceCal;
+            double CalibrationMult = (UseCal ? GetCalibrationMultipler(Humidity, Temperature) : 1.0);
+            Log.Output(Log.Severity.DEBUG, Log.Source.SENSORS, "MQ135 Calculated resistance: " + SensorResistance + ", Ratio: " + ResistanceRatio);
+            if (ResistanceRatio < 0.358 || ResistanceRatio > 2.428) { return -1; }
+            return (float)(116.6020682 * Math.Pow(ResistanceRatio, -2.769034857));
+        }
+
+        // TODO: This still needs to be implemented.
+        // I'm not sure if it will actually be useful.
+        // Math here: http://davidegironi.blogspot.com/2017/07/mq-gas-sensor-correlation-function.html#.WxltRyAh2Un
+        private static double GetCalibrationMultipler(float Humidity, float Temperature)
```

Comment:
I think putting this in will allow us to get much more accurate results since this sensor seems like it would be pretty sensitive, but I don't know how sensitive the sensor is, so I don't know if we have to allocate time to do this right now.